bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/CAPIHelper.java

```diff
@@ -75,6 +75,18 @@ static ByteBuffer toByteBuffer(final mongocrypt_binary_t binary) {
         return pointer.getByteBuffer(0, length);
     }
 
+    static byte[] toByteArray(final mongocrypt_binary_t binary) {
+        ByteBuffer byteBuffer = toByteBuffer(binary);
+        byte[] byteArray = new byte[byteBuffer.remaining()];
+        byteBuffer.get(byteArray);
+        return byteArray;
+    }
+
+    static void writeByteArrayToBinary(final mongocrypt_binary_t binary, byte[] bytes) {
+        Pointer outPointer = mongocrypt_binary_data(binary);
+        outPointer.write(0, bytes, 0, bytes.length);
```

Comment:
Check the mongocrypt_binary_t is large enough for bytes[]

Reply:
Done

Reply:
If this check fails it indicates a bug in libmongocrypt.  But better to report an error in that case than corrupt memory.